ql/cashflows/iborcoupon.cpp

```diff
@@ -86,6 +86,8 @@ namespace QuantLib {
                     accrualEndDate_, -static_cast<Integer>(fixingDays_), Days);
                 fixingEndDate_ = fixingCalendar.advance(
                     nextFixingDate, indexFixingDays, Days);
+                // make sure the estimation period contains at least one day
+                fixingEndDate_ = std::max(fixingEndDate_, fixingValueDate_ + 1);
```

Comment:
The thing that irks me is that this will also be returned by the `fixingEndDate()` method, which might be confusing (or wrong).  Maybe we can introduce another private variable instead?

Reply:
I see, ok. Hmm. But what do we actually expect to get from `fixingEndDate()`? I'd have thought this is the actual end date of the period used for index estimation. Which is a bit artificial in the "par coupon approximation" anyway?

But more importantly this method is used [here](https://github.com/lballabio/QuantLib/blob/master/ql/termstructures/yield/ratehelpers.cpp#L854) and there we'd need the actual end date. Although I'd hope this workaround won't be triggered in the usual rate curve building...

Reply:
Good question.  And the helper needs the adjusted one to avoid extrapolation. Hmm.

Reply:
Maybe the name of the method is not right, if it would be `adjustedFixingEndDate` or something like that it would be clearer that this is not (always) the regular index maturity date?

Reply:
No, I wouldn't add another method for this one unlikely case.  All in all, I think I'll keep your PR as is.

Reply:
Ok I am happy with that. Thanks.